scripts/compute_csa.sh

```diff
@@ -0,0 +1,245 @@
+  # Note: we are using sct_label_utils over sct_label_vertebrae here to avoid straightening (which takes a lot of time)
+  sct_label_utils -i ${FILESEG}.nii.gz -disc ${FILELABEL}.nii.gz -o ${FILESEG}_labeled.nii.gz
+}
+
+
+# Copy GT spinal cord disc labels (located under derivatives/labels)
+copy_gt_disc_labels(){
+  local file="$1"
+  local type="$2"
+  local contrast="$3"
+
+  if [[ $contrast == "T1w" ]] || [[ $contrast == "T2w" ]]; then
+    file_name="${file%%_*}_${contrast}_label-discs_dlabel"
+  elif [[ $contrast == "T2star" ]]; then
+    file_name="${file%%_*}_${contrast}_label-discs_desc-warp_dlabel"
+  elif [[ $contrast == "MTon" ]]; then
+    file_name="${file%%_*}_flip-1_mt-on_MTS_label-discs_desc-warp_dlabel"
+  elif [[ $contrast == "MToff" ]]; then
+    file_name="${file%%_*}_flip-2_mt-off_MTS_label-discs_desc-warp_dlabel"
+  elif [[ $contrast == "DWI" ]]; then
+    file_name="${file%%_*}_rec-average_dwi_label-discs_desc-warp_dlabel"
+  fi
+  # Construct file name to GT segmentation located under derivatives/labels
+  FILEDISCLABELS="${PATH_DATA}/derivatives/labels/${SUBJECT}/${type}/${file_name}.nii.gz"
+  echo ""
+  echo "Looking for manual disc labels: $FILEDISCLABELS"
+  if [[ -e $FILEDISCLABELS ]]; then
+      echo "Found! Copying ..."
+      rsync -avzh $FILEDISCLABELS ${file}_discs.nii.gz
+  else
+      echo "File ${FILEDISCLABELS} does not exist" >> ${PATH_LOG}/missing_files.log
+      echo "ERROR: Manual Disc Labels ${FILEDISCLABELS} does not exist. Exiting."
+      exit 1
+  fi
+}
+
+
+# Copy GT soft binarized segmentation (located under derivatives/labels_softseg_bin)
+copy_gt_softseg_bin(){
+  local file="$1"
+  local type="$2"
+  # Construct file name to GT segmentation located under derivatives/labels_softseg_bin
+  # NOTE: the naming conventions are in the revised BIDS format
+  FILESEG="${PATH_DATA}/derivatives/labels_softseg_bin/${SUBJECT}/${type}/${file}_desc-softseg_label-SC_seg.nii.gz"
+  echo ""
+  echo "Looking for manual segmentation: $FILESEG"
+  if [[ -e $FILESEG ]]; then
+      echo "Found! Copying ..."
+      rsync -avzh $FILESEG ${file}_softseg_bin.nii.gz
+  else
+      echo "File ${FILESEG} does not exist" >> ${PATH_LOG}/missing_files.log
+      echo "ERROR: Manual Segmentation ${FILESEG} does not exist. Exiting."
+      exit 1
+  fi
+}
+
+
+# Segment spinal cord using the contrast-agnostic nnUNet model
+segment_sc_nnUNet(){
+  local file="$1"
+  local file_gt_vert_label="$2"
+  local model_basename="$3"     # 2d or 3d
+  local contrast="$4"   # used only for saving output file name
+  local kernel="$5"    # 2d or 3d_fullres
+
+  FILESEG="${file%%_*}_${contrast}_seg_${model_basename}"
+
+  # Get the start time
+  start_time=$(date +%s)
+  # Run SC segmentation
+  python ${PATH_NNUNET_SCRIPT} -i ${file}.nii.gz -o ${FILESEG}.nii.gz -path-model ${PATH_NNUNET_MODEL}/nnUNetTrainer__nnUNetPlans__${kernel} -pred-type sc -use-gpu -use-best-checkpoint
```

Comment:
I'm curious about the use of this command vs. the use of SCT's `sct_deepseg -custom-url`.

I wonder if it would be possible to write down a list of model URLs (for past versions), then test them all sequentially using SCT? That way you can automate that part of the processing too, rather than manually running the script 5+ times.

Reply:
> I'm curious about the use of this command vs. the use of SCT's `sct_deepseg -custom-url`

so in my understanding, the `-custom-url` flag requires the model to be released already? But here I don't assume that the model has been released, it's still local so that the CSV (which requires to get predictions from the gpu) can be computed and eventually uploaded with the release!

Reply:
Ahhh alright, that makes sense! 

I think I was thinking in the context of comparing with previous releases? But, those only need to be done once, whereas this script will be applied to future iterations as well. :)

Reply:
> But, those only need to be done once,

Indeed! with each successive release, the csa.csv file will be there and the comparison will only fetch these csv files and not rely on `sct_deespeg` functionality anywhere